mongo/integration/unified/collection_operation_execution.go

```diff
@@ -427,6 +424,57 @@ func executeDistinct(ctx context.Context, operation *operation) (*operationResul
 	return newValueResult(bsontype.Array, rawRes, nil), nil
 }
 
+func executeDropIndex(ctx context.Context, operation *operation) (*operationResult, error) {
+	coll, err := entities(ctx).collection(operation.Object)
+	if err != nil {
+		return nil, err
+	}
+
+	var name string
+	dropIndexOpts := options.DropIndexes()
+
+	elems, _ := operation.Arguments.Elements()
+	for _, elem := range elems {
+		key := elem.Key()
+		val := elem.Value()
+
+		switch key {
+		case "name":
+			name = val.StringValue()
+		case "maxTimeMS":
+			dropIndexOpts.SetMaxTime(time.Duration(val.Int32()) * time.Millisecond)
+		}
+	}
+	if name == "" {
+		return nil, newMissingArgumentError("name")
+	}
+
+	res, err := coll.Indexes().DropOne(ctx, name, dropIndexOpts)
```

Comment:
It's probably idiomatic but it seems off to me that we send this one error on with the result rather than through the return arguments.

Reply:
Yeah error returning is slightly confusing in the unified test runner. Many tests in the unified test format expect operations to fail with a particular error. Errors from actually running the operation are returned as the [`Err` field](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/unified/result.go#L27) in the `operationResult`. Only errors from the test runner (like the `newMissingArgumentError` on L449) are returned as the return value for `execute*` functions.